pulsar-function-go/pf/context.go

```diff
@@ -119,6 +118,15 @@ func (c *FunctionContext) GetUserConfMap() map[string]interface{} {
 	return c.userConfigs
 }
 
+func (c *FunctionContext) GetSecretsMap() map[string]interface{} {
```

Comment:
I don't think it is a good idea to pass `secrets` in plain text. In Java & Python functions, we have introduced a notion called a secrets provider. The secrets provider is responsible for retrieving secrets for functions. 

I'd suggest moving the secrets change to a separate pull request. Because it requires a fair amount of work to get a good implementation with a secrets provider.

Reply:
So, according to the setup code I looked at, the secrets are already getting sent into the function context, just not actually referenced anywhere. Is the concern that the existing method doesn't allow for a pluggable `SecretsProvider` interface? If so, that makes sense and I'll amend the PR and we can address it later.

Reply:
@flowchartsman As we discussed, we can separate this pr and provide the secretsProvider interface to allow users to customize their own encryption methods. By default, we can provide an implementation of reading secrets from environment variables.